components/validation/src/main/java/org/wso2/carbon/identity/x509Certificate/validation/service/CertificateValidationManagementServiceImpl.java

```diff
@@ -0,0 +1,219 @@
+import org.wso2.carbon.identity.x509Certificate.validation.CertificateValidationException;
+import org.wso2.carbon.identity.x509Certificate.validation.constant.error.ErrorMessage;
+import org.wso2.carbon.identity.x509Certificate.validation.exception.X509ConfigurationException;
+import org.wso2.carbon.identity.x509Certificate.validation.model.CACertificate;
+import org.wso2.carbon.identity.x509Certificate.validation.model.CACertificateInfo;
+import org.wso2.carbon.identity.x509Certificate.validation.model.CertObject;
+import org.wso2.carbon.identity.x509Certificate.validation.model.Validator;
+import org.wso2.carbon.identity.x509Certificate.validation.util.X509ConfigurationExceptionHandler;
+
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.List;
+
+import static org.wso2.carbon.identity.configuration.mgt.core.constant.ConfigurationConstants.ErrorMessages.ERROR_CODE_RESOURCES_DOES_NOT_EXISTS;
+import static org.wso2.carbon.identity.x509Certificate.validation.CertificateValidationUtil.addCertificateInConfigurationStore;
+import static org.wso2.carbon.identity.x509Certificate.validation.CertificateValidationUtil.deleteCACertificateFromCertificateManager;
+import static org.wso2.carbon.identity.x509Certificate.validation.CertificateValidationUtil.deleteCertificateInConfigurationStoreByCertificateId;
+import static org.wso2.carbon.identity.x509Certificate.validation.CertificateValidationUtil.getCertificateFromConfigurationStoreByCertificateId;
+import static org.wso2.carbon.identity.x509Certificate.validation.CertificateValidationUtil.getCertificateListFromConfigurationStore;
+import static org.wso2.carbon.identity.x509Certificate.validation.CertificateValidationUtil.getNormalizedName;
+import static org.wso2.carbon.identity.x509Certificate.validation.CertificateValidationUtil.getValidatorFromConfigStoreByName;
+import static org.wso2.carbon.identity.x509Certificate.validation.CertificateValidationUtil.getValidatorsFromConfigStore;
+import static org.wso2.carbon.identity.x509Certificate.validation.CertificateValidationUtil.updateCACertificateInCertificateManager;
+import static org.wso2.carbon.identity.x509Certificate.validation.CertificateValidationUtil.updateCertificateInConfigurationStoreByCertificateId;
+import static org.wso2.carbon.identity.x509Certificate.validation.CertificateValidationUtil.updateValidatorInConfigStore;
+
+/**
+ * This implementation handles the x509 authenticator validator manager implementation.
+ */
+public class CertificateValidationManagementServiceImpl implements CertificateValidationManagementService {
+
+    @Override
+    public List<Validator> getValidators(String tenantDomain) throws X509ConfigurationException {
+
+        try {
+            return getValidatorsFromConfigStore(tenantDomain);
+        } catch (ConfigurationManagementException e) {
+            if (ERROR_CODE_RESOURCES_DOES_NOT_EXISTS.getCode().equals(e.getErrorCode())) {
```

Comment:
Can't we pass an empty array without throwing an exception

Reply:
fixed with https://github.com/wso2-extensions/identity-x509-commons/pull/57/commits/52f92488d265780c25c0d23c5ec638ca00bd3f59